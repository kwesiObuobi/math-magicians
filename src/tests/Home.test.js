import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

describe('Home Component', () => {
  it('renders the Home component', () => {
    render(<Home />);
    expect(screen.getByText(/Welcome/)).toBeInTheDocument();
    expect(screen.getByText(/The introduction/)).toBeInTheDocument();
    expect(screen.queryByRole('link')).toBeNull();
  });

  it('should correctly render the Home component', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
