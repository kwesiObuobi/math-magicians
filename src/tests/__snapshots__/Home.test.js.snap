// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should correctly render the Home component 1`] = `
<div
  className="home-container"
>
  <h1
    className="home-header"
  >
    Welcome to our page!
  </h1>
  <p
    className="home-text"
  >
    The introduction of calculators marked a significant milestone in the history of mathematics. Before the advent of calculators, mathematical calculations were performed using slide rules and mechanical calculators, which were often cumbersome and slow. The first electronic calculator was invented by Jack Kilby in 1967, and it could perform basic arithmetic operations. However, early calculators were bulky, expensive, and not very user-friendly. Over time, advances in technology led to the development of more advanced calculators that could perform complex mathematical calculations quickly and efficiently. Today, calculators are an essential tool in many fields, including science, engineering, finance, and education.
  </p>
  <p
    className="home-text"
  >
    They have become more compact, affordable, and user-friendly, and they have features such as graphing, programming, and statistical analysis. The widespread availability of calculators has made it easier for people to perform calculations, and it has also led to the development of new mathematical tools and techniques. However, some argue that the overreliance on calculators has led to a decline in basic math skills, and it is still important for people to understand the underlying principles of mathematics. Overall, the introduction of calculators has had a profound impact on the field of mathematics, and it will continue to play an important role in the years to come.
  </p>
</div>
`;
